packages/url/src/index.js

```diff
@@ -167,13 +167,14 @@ export function isValidPath( path ) {
  * @example
  * ```js
  * const queryString1 = getQueryString( 'http://localhost:8080/this/is/a/test?query=true#fragment' ); // 'query=true'
- * const queryString2 = getQueryString( 'https://wordpress.org#fragment?query=false&search=hello' ); // 'query=false&search=hello'
+ * const queryString2 = getQueryString( 'https://wordpress.org#fragment?query=false&search=hello' ); // undefined
  * ```
  *
  * @return {?string} The query string part of the URL.
  */
 export function getQueryString( url ) {
-	const matches = /^\S+?\?([^\s#]+)/.exec( url );
+	const cleanUrl = removeFragment( url );
+	const matches = /^\S+?\?([^\s#]+)/.exec( cleanUrl );
```

Comment:
This regular expression seems to imply that it's matching all characters up to a hash `#`, which seems like at the very least the proposed changes make that now unnecessary for consideration.

```suggestion
	const matches = /^\S+?\?([^\s]+)/.exec( cleanUrl );
```